core-runner/src/runner.rs

```diff
@@ -313,7 +305,18 @@ impl<SC: StorageCarrier, E: Environment<Ext>> Runner<SC, E> {
             self.block_info,
         );
 
+        let outgoing_messages = context.message_buf.drain(..).collect::<Vec<_>>();
+
         if run_result.outcome.was_trap() && generate_reply_on_trap {
+            let gas_spent_for_outgoing: u64 =
+                outgoing_messages.iter().map(|msg| msg.gas_limit).sum();
```

Comment:
Grumble: `map` + `sum` may be replaced by `fold`.

Reply:
but it requires to write summation function by hand